x-pack/legacy/plugins/code/server/repository_service.ts

```diff
@@ -93,6 +93,9 @@ export class RepositoryService {
   public async remove(uri: string): Promise<DeleteWorkerResult> {
     const localPath = RepositoryUtils.repositoryLocalPath(this.repoVolPath, uri);
     try {
+      if (localPath.includes('../')) {
```

Comment:
startWith(../) || includes('../)

also, should we consider windows path?

Reply:
good catch. changed to `localPath.split(path.sep).includes('..')` where `path.sep` could be window compaitble.

For the other 2 places, because mostly they are RepositoryUri which use `/` as separator even on windows, then split by `/` should be good.